### ISSUE TITLE
Cop: SketchupBugs/MaterialName

### ISSUE THREAD

**Issue by thomthom** [SketchUp/rubocop-sketchup#87]
`materials.add_material` and `material.name=` was bugged in earlier versions.

In SU2017 and earlier it allowed for duplicate names to be set.
In SU2018 it raised an error when setting material name to itself, and had a name cache bug

**Comment by thomthom** [comment]
It's a challenge to figure out a reliable way to flag this. `add_material` is probably fine. But `material.name=` is trickier. Looking for `#name=` is too generic and cause too many false positives. Might have to do some naive assumptions of common variable names: `material`, `mat` etc...

Then there is the consideration of accounting for fixes to this - the API user using `unique_name`.

**Comment by thomthom** [comment]
```ruby
# This can cause duplicate material names in SU2017 and earlier.
materials.add_material('Example')
```

---

```ruby
# Works in SU2018+
name = materials.unique_name('Example')
material.name = name
```

```ruby
# Works in SU2018+
material.name = materials.unique_name('Example')
```

```ruby
# Might raise ArgumentError in SU2018+ if name is not unique
material.name = 'Example'
```

**Comment by thomthom** [comment]
Fixed in 2fafa248048cd25600c7d1b6a5ce155413f5fb11
